### PR TITLE
machineConfig: Add WithRawConfig method

### PR DESCRIPTION
**Pull request by ajaggapa**

### CODE REVIEW

**Review by sebrandon1 — APPROVED** [review]

File: pkg/mco/machineconfig.go [hunk 1]
```diff
@@ -327,6 +327,27 @@ func (builder *MCBuilder) WithKernelType(kernelType string) *MCBuilder {
...
+
+	if len(config) == 0 {
+		glog.V(100).Infof("The Config.Raw cannot be empty")
+
+		builder.errorMsg = "'Config.Raw' cannot be empty"
+
+		return builder
+	}
+
+	glog.V(100).Infof("Setting Config.Raw: %v", config)
```
Comment (evgenLevin) [inline pkg/mco/machineconfig.go:344]: Will it be human-readable?

Comment (ajaggapa) [reply pkg/mco/machineconfig.go:344]: It is a string converted to []byte. So it is reversible.

Comment (kononovn) [reply pkg/mco/machineconfig.go:344]: I see we have Raw field. Can you share link to documentation of this field

Comment (ajaggapa) [reply pkg/mco/machineconfig.go:344]: In my use case, i want to have ignition config to be passed to machineconfig. Below is code on how i use it.

```
import ignition "github.com/coreos/ignition/v2/config/v3_1/types"
....
var (
	err                       error
	RdmaMachineConfigContents = `

      [Unit]
      Description=RDMA exclusive mode
      Before=kubelet.service crio.service node-valid-hostname.service

      [Service]
      # Need oneshot to delay kubelet
      Type=oneshot
      ExecStart=/usr/bin/bash -c "rdma system set netns exclusive"
      StandardOutput=journal+console
      StandardError=journal+console

      [Install]
      WantedBy=network-online.target`
)

truePointer := true

ignitionconfig := ignition.Config{
	Ignition: ignition.Ignition{
		Version: "3.1.0",
	},
	Systemd: ignition.Systemd{
		Units: []ignition.Unit{
			{
				Enabled:  &truePointer,
				Name:     "rdma.service",
				Contents: &RdmaMachineConfigContents,
			},
		},
	},
}

ignitionconfigraw, err := json.Marshal(ignitionconfig)

rdmaMachineConfig := mco.NewMCBuilder(clients.New(kubeconfigpath), "02-rdma-netns-exclusive-mode").WithLabel("machineconfiguration.openshift.io/role", "workercnf").WithRawConfig(ignitionconfigraw)
```

Comment (ajaggapa) [reply pkg/mco/machineconfig.go:344]: is there any other better way to implement this machine config ?

Comment (ajaggapa) [reply pkg/mco/machineconfig.go:344]: https://docs.openshift.com/container-platform/4.16/rest_api/machine_apis/machineconfig-machineconfiguration-openshift-io-v1.html#spec

Comment (ajaggapa) [reply pkg/mco/machineconfig.go]: @trewest 
Agreed. Updated to `glog.V(100).Infof("Setting Config.Raw: %s", string(config))`

File: pkg/mco/machineconfig_test.go [hunk 2]
```diff
@@ -394,6 +394,32 @@ func TestMachineConfigWithExtensions(t *testing.T) {
 	}
 }
 
+func TestWithRawConfig(t *testing.T) {
```
Comment (kononovn) [inline pkg/mco/machineconfig_test.go:397]: Please re-name:
TestMachineConfigWithRawConfig

**Review by kononovn — CHANGES_REQUESTED** [review]
1 change required

File: pkg/mco/machineconfig_test.go (hunk 2 above)
Comment (ajaggapa) [reply pkg/mco/machineconfig_test.go]: Updated 👍

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**